pkg/natdiscovery/response_handle.go

```diff
@@ -0,0 +1,83 @@
+/*
+© 2021 Red Hat, Inc. and others
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package natdiscovery
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/pkg/errors"
+	"k8s.io/klog"
+
+	"github.com/submariner-io/submariner/pkg/natdiscovery/proto"
+)
+
+func (nd *natDiscovery) handleResponseFromAddress(req *proto.SubmarinerNatDiscoveryResponse, addr *net.UDPAddr) error {
+	if req.GetSender() == nil || req.GetReceiver() == nil || req.GetReceivedSrc() == nil {
+		return errors.Errorf("received malformed response")
```

Comment:
Shall we include the req in the error message to aid in debugging?

Reply:
Ack. Done.